pkg/apis/admission/types.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admission
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/kubernetes/pkg/apis/authentication"
+)
+
+// AdmittanceReview describes an admission request.
+type AdmittanceReview struct {
+	metav1.TypeMeta
+
+	// Spec describes the attributes for the admission request.
+	Spec AdmittanceReviewSpec
+	// Status is filled in by the webhook and indicates whether the admission request should be permitted.
+	Status AdmittanceReviewStatus
+}
+
+// AdmittanceReviewSpec describes the admission.Attributes for the admission request.
+type AdmittanceReviewSpec struct {
+	// Kind is the type of object being manipulated.  For example: Pod
+	Kind metav1.GroupVersionKind
+	// Name is the name of the object as presented in the request.  On a CREATE operation, the client may omit name and
+	// rely on the server to generate the name.  If that is the case, this method will return the empty string.
+	Name string
+	// Namespace is the namespace associated with the request (if any).
+	Namespace string
+	// Object is the object from the incoming request prior to default values being applied
+	Object runtime.Object
+	// OldObject is the existing object. Only populated for UPDATE requests.
```

Comment:
Not about mutating or not, just because of performance.  The mutating field would be returned in status.  That way a caller only has to check allowed, and optionally a response, if and only if it was changed.